science/limesuite/Portfile

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.0
+PortGroup           cxx11 1.1
+PortGroup           github 1.0
+PortGroup           wxWidgets 1.0
+
+name                limesuite
+version             v19.01.0
+categories          science comms
+platforms           darwin macosx
+license             Apache-2
+maintainers         {@ra1nb0w irh.it:rainbow} openmaintainer
+description         provides drivers and SDR application support for the LMS7002M RFIC
+long_description    Lime Suite is a collection of software supporting several hardware \
+    platforms including the LimeSDR, drivers for the LMS7002M transceiver RFIC, and other \
+    tools for developing with LMS7-based hardware.
+homepage            https://myriadrf.org/projects/lime-suite/
+github.setup        myriadrf LimeSuite 025ffa1a1026a6307b001ea2a6ffb9ab5c4e8d63
+
+checksums           rmd160  dbb56b3657d3e7fcdcda50e137400efd7f0d3417 \
+                    sha256  e7228c53d5fbe93a490d82fdcb94a54fb9a1d5cacc0bbf2c581609e859f3cd86 \
+                    size    5347698
+
+depends_lib-append  port:libusb \
+                    port:sqlite3
+
+configure.args-append \
+    -DDOWNLOAD_IMAGES=OFF \
+    -DENABLE_STREAM=ON \
+    -DENABLE_GUI=OFF \
+    -DENABLE_NOVENARF7=OFF \
+    -DENABLE_SOAPY_LMS7=OFF\
+    -DENABLE_OCTAVE=OFF
+
+# need 3.2 to enable dark mode in mojave
+variant wxgui description {Enable LimeSuiteGUI} {
+    wxWidgets.use           wxWidgets-3.2
+    depends_lib-append      port:${wxWidgets.port}
+    configure.args-delete   -DENABLE_GUI=OFF
+    configure.args-append   -DENABLE_GUI=ON
+}
+
+variant soapy description {Add Soapy support} {
+    depends_lib-append      port:SoapySDR
+    configure.args-delete   -DENABLE_SOAPY_LMS7=OFF
+    configure.args-append   -DENABLE_SOAPY_LMS7=ON
+}
+
+variant download_images description {Download and install firmware/gatware images} {
+    configure.args-delete   -DENABLE_STREAM=OFF
+    configure.args-append   -DENABLE_STREAM=ON
+}
```

Comment:
Is the download for this large enough that someone would want to skip this?

Reply:
More importantly: Aren't these images required to use any SDR HW that uses LimeSuite? Can one use LimeSuite without SDR HW ... meaning is it like GNU Radio in that the HW is provided by speciality modules? I'm thinking not but I honestly don't know since I have no SDR HW that works with this project (yet).

Reply:
The download is nearly 4.5MB. If you don't enable that variant you can update the gatware at runtime with LimeSuiteGUI or `LimeUtil --update` and wget as dependency. Generally use case, seeing Pothos, is to use runtime.
No, those images are not necessary to use LimeSuite but it is suggested to upgrade it every new version because ABI changes.
No, LimeSuite works as "controller" of LMS7002M RFIC boards, like limesuite, limesuite-mini, stream and others, so you need the board to do something useful. When this port will be in the main repo, I send gr-limesdr that it is a module for use limesdr in GNU Radio, as sink or source.

Reply:
Hmmm ... interesting & OK.

What I do for UHD is in `post-destroot` have a UHD utility download and install the images into the `destroot` area but into a subdirectory where, once activated, UHD will search for them. UHD images are specific to a release / commit; some change a lot between releases while others (old SDR HW) generally don't. I find it better to have MP download and install the images instead of pushing this off to the user: UHD requires them for runtime && this method is reliable as we can direct the images to a location we know UHD runtime will look for them in. If we allow the user to dictate where the images are downloaded, it makes debugging issues much harder for us.

Also: 4.5 MB isn't that much data.

Questions:
+ Are LimeSuite images specific to a given release / commit?
+ Can LimeSuite images change for a given release / commit?
+ Is `LimeUtil` provided by LimeSuite?
+ Although you note that the images aren't necessary to use LimeSuite, if LimeSuite is just a controller of various SDR HW then -somewhere- in the process of executing LimeSuite the images will be required to load onto the HW to get it going.

My vote is to add in the `post-destroot` phase to download and install the images. In this manner, we can provide the user with a simple and easy experience getting LimeSuite up and running. But, if you truly feel otherwise I'm still open to discussion.

Reply:
> What I do for UHD is in `post-destroot` have a UHD utility download and install the images into the `destroot` area but into a subdirectory where, once activated, UHD will search for them. UHD images are specific to a release / commit; some change a lot between releases while others (old SDR HW) generally don't. I find it better to have MP download and install the images instead of pushing this off to the user: UHD requires them for runtime && this method is reliable as we can direct the images to a location we know UHD runtime will look for them in. If we allow the user to dictate where the images are downloaded, it makes debugging issues much harder for us.

LimeSuite already try to find the images in the following folders:
`Image search paths:
     - /Users/rainbow/.local/share/LimeSuite/images
     - /opt/local/share/LimeSuite/images
`
if not found it starts wget. In fact, if you use `+download_images` cmake download the images in `share/LimeSuite/Images/${version}` where `${version}` now is 19.01

> * Are LimeSuite images specific to a given release / commit?

From my little experience, yes they are specific per release.

> * Can LimeSuite images change for a given release / commit?

I didn't see that, but my experience is long only a few months

> * Is `LimeUtil` provided by LimeSuite?

yes

> * Although you note that the images aren't necessary to use LimeSuite, if LimeSuite is just a controller of various SDR HW then -somewhere- in the process of executing LimeSuite the images will be required to load onto the HW to get it going.

It is a controller of RFIC configurations and the images are flashed into fpga ad rfic/flash.
Hopping to understand correctly your point, a typical use case is:

- receive the new board (limesdr in my case)
- run `LimeSuiteGUI` and it warns if your version is not aligned with the installed limesuite package; then you can decide if upgrade or not
- configure limesdr from the LimeSuite
- you can also avoid to open `LimeSuiteGUI` or `LimeUtil` and start to use it from GNU Radio or Pothos. For the first one, you generally need to get the serial with `LimeUtil --find`. If you don't update, you could fall in an ABI mismatch.

> My vote is to add in the `post-destroot` phase to download and install the images. In this manner, we can provide the user with a simple and easy experience getting LimeSuite up and running. But, if you truly feel otherwise I'm still open to discussion.

I didn't checked how much space `download_images` occupy and 4.5MB is not too much.
So, at this point, it is a good idea to enable `download_images` as default. Do you see some trouble to let cmake download the images?

Reply:
Yeah that makes sense: let cmake download the images. Let's not provide a variant for downloading images, and instead just enable it always.

Reply:
Ok. Note: it downloads also three dll and a zip with GNU Radio block (<2MB). To modify this behaviour the port need to patch `cmake/DownloadImages.cmake` L18. Leave it or replace the regex?

Reply:
2 options: (1) patch the file to handle the regex to just be for the needed images. (2) don't patch; download all images & then in post-build or post-destroot delete the unwanted images. Either can work, and, honestly, (2) is probably preferred even though it wastes some bandwidth & time ...

Reply:
Followed the second one. I used `file delete` with `glob`; is it the right way?

Reply:
`glob` looks OK to me. I'll test to verify.